os/hal/lib/streams/chprintf.c

```diff
@@ -79,9 +80,32 @@ static const long pow10[FLOAT_PRECISION] = {
     10, 100, 1000, 10000, 100000, 1000000, 10000000, 100000000, 1000000000
 };
 
-static char *ftoa(char *p, double num, unsigned long precision) {
+
+// rusEFI
+#if defined(__cplusplus) && defined(__OPTIMIZE__)
+#include <type_traits>
+// "g++ -O2" version, adds more strict type check and yet no "strict-aliasing" warnings!
+#define rusefi_cisnan(f) ({ \
+	static_assert(sizeof(f) == sizeof(int32_t)); \

```

Comment:
de we need the c++ case in a c file?

Reply:
Had either of us go to a boarding school somewhere in East Asia?